concreteproperties/design_codes.py

```diff
@@ -727,3 +739,1677 @@ def biaxial_bending_diagram(
+        """Modification factor reflecting the reduced mechanical properties of
+        lightweight concrete relative to normal weight concrete of the same compression
+        strength.
+
+        :math:`\quad\lambda=0.4+\displaystyle{\\frac{0.6\\rho}{2200}}\\leq1.0`
+
+        :param density: Saturated surface dry density of concrete material
+        :return: :math:`\lambda` factor
+        """
+        lamda = min(0.4 + 0.6 * density / 2200, 1)
+        return lamda
+
+    def concrete_capacity(
+        self,
+        os_design: bool = False,
+        prob_design: bool = False,
+        add_compressive_strength: float = 15,
+    ) -> float:
+        """Function to return the nominal concrete capacity or the overstrength concrete
+        capacity of a concrete section.
+
+        :param os_design: True if an overstrength capacity of a concrete section is
+            required, then the material properties for concrete are scaled to reflect
+            the likely maximum material strength properties
+        :param prob_design: True if the probable capacity of a concrete section is
+            required, then the material properties for concrete and lumped reinforcement
+            are scaled to reflect the probable material strength properties
+        :param add_compressive_strength: The increase in compressive strength of the
+            specified 28 day compressive strength of concrete to reflect the likely
+            maximum material strength, defaults to an additional 15 MPa as per
+            NZS3101:2006 CL 2.6.5.5(c)
+        :return: Nominal, overstrength or probable concrete yield force (N)
+        """
+        # initiate force variable
+        force = 0
+
+        # loop through all concrete geometries
+        for conc_geom in self.concrete_section.concrete_geometries:
+            # calculate concrete area & compressive strength
+            concrete_area = conc_geom.calculate_area()
+            compressive_strength = (
+                conc_geom.material.ultimate_stress_strain_profile.get_compressive_strength()
+            )
+
+            # scale concrete compressive strength for overstrength if specified
+            if prob_design:
+                mult_compressive_strength = 1.5 if compressive_strength <= 40 else 1.4
+                compressive_strength *= mult_compressive_strength
+            elif os_design:
+                compressive_strength += add_compressive_strength
+
+            # calculate cumulative concrete force
+            force += (
+                self.alpha_1(compressive_strength)
+                * concrete_area
+                * compressive_strength
+            )
+
+        return force
+
+    def steel_capacity(
+        self, os_design: bool = False, prob_design: bool = False
+    ) -> float:
+        """Function to return the nominal steel reinforcement capacity or the
+            overstrength steel reinforcement capacity of a concrete section.
```

Comment:
I think there is an unintentional tab here, docs formatting is thrown by this a bit.

Reply:
yes missed this, have corrected.